src/high_level.rs

```diff
@@ -622,6 +622,47 @@ pub fn exec_cell(
     syscalls::exec_cell(code_hash, hash_type, offset, length, argv)
 }
 
+/// Locate a cell in cell deps and load the cell data for execution.
+///
+/// # Arguments
+///
+/// * `code_hash` - the code hash to search cell in cell deps.
+/// * `hash_type` - the hash type to search cell in cell deps.
+/// * `offset`    - the offset of the cell data to load.
+/// * `length`    - the length of the cell data to load, if the length is 0, load the whole cell data starting from offset.
+/// * `args`      - the arguments to pass to the execution.
+pub fn exec_cell_with_args(
+    code_hash: &[u8],
+    hash_type: ScriptHashType,
+    offset: u32,
+    length: u32,
+    args: &[u8],
```

Comment:
This is very different from the RFC or how args are passed in C/POSIX. We need to document the exact mechanism and how to recover it on the callee side if we are really going to add this.